framework/validators/Validator.php

```diff
@@ -91,9 +91,9 @@ class Validator extends Component
         'safe' => 'yii\validators\SafeValidator',
         'string' => 'yii\validators\StringValidator',
         'trim' => [
-            'class' => 'yii\validators\FilterValidator',
-            'filter' => 'trim',
+            'class' => 'yii\validators\TrimValidator',
             'skipOnArray' => true,
```

Comment:
With skipOnArray set to true in the class itself we can remove this overriden config and just leave `'trim' => 'yii\validators\TrimValidator'`

Reply:
it's change default behavior

Reply:
How? skippOnArray is set to true here so I want to put it on the class level to not change the behavior.

Reply:
@bizley `TrimValidator` behavior must be same to `FilterValidator`
https://github.com/yiisoft/yii2/blob/7f244793f561f770f2945e5d4be91f08eecaf8c7/framework/validators/FilterValidator.php#L52

Reply:
Jesus, man, you're the worst. From the beginning `trim` had the `skipOnArray` option set to `true` but now you say that somehow it's behavior **must** be the same as FilterValidator. Ok, let it be your way, but remove the `'skipOnEmpty' => false,` then since it's the same as in the default config.

Reply:
@bizley
> Ok, let it be your way, but remove the 'skipOnEmpty' => false, then since it's the same as in the default config.

what's you mean?

https://github.com/yiisoft/yii2/blob/7f244793f561f770f2945e5d4be91f08eecaf8c7/framework/validators/FilterValidator.php#L57
https://github.com/yiisoft/yii2/blob/7f244793f561f770f2945e5d4be91f08eecaf8c7/framework/validators/Validator.php#L139

Reply:
@bizley the user could override the `trim` сonfig, he can omit `skipOnArray` to enable it:
~~~php
'trim' => [
    'class' => 'yii\validators\FilterValidator',
    'filter' => 'trim',
],
~~~
and if user replace only class in custom config that change his default behaviour.
~~~php
'trim' => [
    'class' => 'yii\validators\TrimValidator',
],
~~~

Reply:
I mean:
```
'trim' => [
            'class' => 'yii\validators\TrimValidator',
            'skipOnArray' => true,
            'skipOnEmpty' => false,
        ],
```
and yet:
```
public $skipOnArray = false;
```
because this somehow must be the same as in FilterValidator while we change it anyway in the config above and while it was set like that for the last thousand years,
and the thing you asked:
```
public $skipOnEmpty = false;
```
which is the same as in the config above so it's not needed in the config as I said. Is it clear now?

Reply:
@bizley didn't understand what's you told about `$skipOnEmpty`

Reply:
@bizley aa.. now understand you: `'skipOnEmpty' => false,` it's really my fail